src/routes.js

```diff
@@ -316,3 +315,27 @@ export default (apollo, store) => {
     }
   ]
 }
+
+// handle direct sign in from the extension via deeplink
+export async function extensionSignIn({ to, next }, apollo, store) {
+  const {
+    data: { network }
+  } = await apollo.query({
+    query: gql`
+      query Network {
+        network(id: "${to.params.network}") {
+          id
+          slug
+        }
+      }
+    `,
+    fetchPolicy: "cache-first"
+  })
+
+  await store.dispatch(`signIn`, {
+    sessionType: `extension`,
+    address: to.params.address,
+    networkId: to.params.network
+  })
+  next(`/${network.slug}/portfolio`)
+}
```

Comment:
This side change with the routing should be another PR no? ;)

Or why is it needed here?